File: 백준/Silver/8911. 거북이/거북이.py
import sys
input = sys.stdin.readline

di = [0, 1, 0, -1] # 0: 동 1: 남 2: 서 3: 북
dj = [1, 0, -1, 0]
dir = 0
T = int(input())

for i in range(T):
    commend = input().strip()
    start = [0, 0]
    min_x, max_x, min_y, max_y = 0, 0, 0, 0
    
    for c in commend:
        if c == 'F':
            start[0] += di[dir]
            start[1] += dj[dir]
        elif c == 'B':
            start[0] -= di[dir]
            start[1] -= dj[dir]
        elif c == 'L':
            dir = (dir - 1) % 4
        elif c == 'R':
            dir = (dir + 1) % 4
            
        min_x = min(min_x, start[0])
        max_x = max(max_x, start[0])
        min_y = min(min_y, start[1])
        max_y = max(max_y, start[1])
        
    print(abs(max_x - min_x) * abs(max_y - min_y))
        

File: 백준/Silver/8911. 거북이/거북이.js
const fs = require("fs");
// let input = fs.readFileSync("./input.txt").toString().trim().split("\n");
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const di = [-1, 0, 1, 0]
const dj = [0, 1, 0, -1]
let dir = 0
const T = parseInt(input[0])

for (let i=1; i <= T; i++) {
  let commend = input[i]
  let start = new Array(2).fill(0)
  let min_x = 0, max_x = 0, min_y = 0, max_y = 0;

  for (let c of commend) {
    if (c === 'F') {
      start[0] += di[dir]
      start[1] += dj[dir]
      } else if (c === 'B') {
        start[0] -= di[dir]
        start[1] -= dj[dir]  
    } else if (c === 'L') {
      dir = (dir - 1 + 4) % 4
    } else if (c === 'R') {
      dir = (dir + 1) % 4
    }

    min_x = Math.min(min_x, start[0])
    max_x = Math.max(max_x, start[0])
    min_y = Math.min(min_y, start[1])
    max_y = Math.max(max_y, start[1])
  }

  console.log(Math.abs(max_x - min_x) * Math.abs(max_y - min_y))
}